src/morphology/german/getForms.js

```diff
@@ -205,29 +105,41 @@ const addFormsWithRemovedLetters = function( morphologyDataNouns, stemmedWordToC
 export function getForms( word, morphologyData ) {
 	const stemmedWord = stem( word );
 	const forms = new Array( word );
-	const exceptions = checkExceptions( morphologyData.nouns, stemmedWord );
 
-	// Check exceptions.
+	/*
+	 * Check whether the word is on an exception list. Since a given stem might sometimes be on an exception list in
+	 * different word categories (e.g., "sau-" from the noun "Sau" or the adjective "sauer") we need to do this cumulatively.
+	 */
```

Comment:
I would change this comment to reflect that the function calls below also generate the forms.